jupyterhub_moss/form/option_form.js

```diff
@@ -566,21 +574,25 @@ document.addEventListener('DOMContentLoaded', () => {
     .addEventListener('change', (e) => updateMemValue());
 
   // Handle add custom environment
-  document
-    .getElementById('environment_add_name')
-    .addEventListener('input', (e) => {
-      const isInputEmpty = e.target.value === '';
-      if (environmentAddRadio.value === '' && !isInputEmpty) {
-        // First input in the name: select its radio button
-        environmentAddRadio.checked = true;
-      }
-      environmentAddRadio.value = e.target.value;
-      environmentAddButton.disabled = isInputEmpty;
-    });
+  environmentAddPath.addEventListener('input', updateAddCustomEnvironment);
+  environmentAddMods.addEventListener('input', updateAddCustomEnvironment);
+
+  environmentAddName.addEventListener('focus', () => {
+    environmentAddRadio.checked = true;
+    updateAddCustomEnvironment();
+  });
+  environmentAddPath.addEventListener('focus', () => {
+    environmentAddRadio.checked = true;
+    updateAddCustomEnvironment();
+  });
+  environmentAddMods.addEventListener('focus', () => {
+    environmentAddRadio.checked = true;
+    updateAddCustomEnvironment();
+  });
```

Comment:
Select radio button on `focus` rather than on "first" `input`. This makes it straight forward.